src/main/java/christmas/controller/ChristmasController.java

```diff
@@ -0,0 +1,148 @@
+        return new Orders(orders);
+    }
+
+    private ExpectedVisitDate readDate(int date) {
+        return new ExpectedVisitDate(date);
+    }
+
+    private BeforeEventPrice getBeforeEventPrice(Orders orders) {
+        BeforeEventPrice beforeEventPrice = new BeforeEventPrice(orders);
+        printBeforeEventPrice(beforeEventPrice);
+        return beforeEventPrice;
+    }
+
+    private void printGift(BeforeEventPrice beforeEventPrice) {
+        Gift gift = new Gift(beforeEventPrice.getBeforePrice(), Menu.CHAMPAGNE.getName(), GIFT_GIVEN_COUNT.getNumber());
+        printGift(gift);
+    }
+
+    private Benefits getBenefits(Orders orders, ExpectedVisitDate date) {
+        Benefits benefits = new Benefits(orders.toDtoList(), date);
+        printBenefits(benefits);
+        return benefits;
+    }
+
+    private BenefitSum getBenefitSum(Benefits benefits) {
+        BenefitSum benefitSum = new BenefitSum(benefits);
+        printBenefitSum(benefitSum);
+        return benefitSum;
+    }
+
+    private void printAfterEventPrice(BeforeEventPrice beforeEventPrice, Benefits benefits) {
+        AfterEventPrice afterEventPrice = new AfterEventPrice(beforeEventPrice, benefits);
+        printAfterEventPrice(afterEventPrice);
+    }
+
+    private void printBadge(BenefitSum benefitSum) {
+        Badge badge = getBadge(benefitSum.getBenefitSum().getSum());
+        printBadge(badge);
+    }
+
+    private void printGift(Gift gift) {
+        if (gift.getCount().getCount() <= 0) {
+            outputView.printGift(GIFT_NOTHING.getString());
+        }
+        if (gift.getCount().getCount() > 0) {
+            outputView.printGift(PRINT_ORDER_FORMAT.getString().formatted(
+                    MessageFormat.format(ORDER_PRINT_FORMAT.getString(), gift.getName().getName(),
+                            gift.getCount().getCount())));
+        }
+    }
+
+    private void printBeforeEventPrice(BeforeEventPrice beforeEventPrice) {
+        outputView.printBeforePrice(beforeEventPrice.getBeforePrice());
+    }
+
+    private void printBenefitSum(BenefitSum benefitSum) {
+        int sum = ZERO.getNumber();
+        if (benefitSum.getBenefitSum().getSum() > ZERO.getNumber()) {
+            sum = -benefitSum.getBenefitSum().getSum();
+        }
+        outputView.printBenefitSum(sum);
+    }
+
+    private void printAfterEventPrice(AfterEventPrice afterEventPrice) {
+        outputView.printAfterEventPrice(afterEventPrice.getAfterEventPrice());
+    }
+
+    private void printBenefits(Benefits benefits) {
+        outputView.printBenefits(benefits.getBenefits().printBenefits());
+    }
+
+    private Badge getBadge(int benefitSum) {
+        if (benefitSum >= Badge.SANTA.getCost()) {
+            return Badge.SANTA;
+        }
+        if (benefitSum >= Badge.TREE.getCost()) {
+            return Badge.TREE;
+        }
+        if (benefitSum >= Badge.STAR.getCost()) {
+            return Badge.STAR;
+        }
+        return Badge.NONE;
+    }
```

Comment:
Badge를 고르는 책임을 enum 내부로 옮겨주면 좋을 것 같아요. 람다식을 활용해보세요.

```suggestion
    private Badge getBadge(int benefitSum) {
        return Badge.getAvailableBadge(benefitSum);
    }
```